### PR TITLE
testing recipe controller

### PR DESCRIPTION
**Pull request by SamiDghim**
generate new controller for recipe

switch to resourceful route

add some comment to recipe_controller

add controller_marcos.rb to test authentification

add test for user model

add database_cleaner to clean testing database

test authentification and get the current user

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe User, :type => :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (moelheni) [inline spec/models/user_spec.rb]: Delete it if you're not going to add tests

File: app/views/recipe/create.html.haml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+%h1 Recipe#create
```
Comment (moelheni) [inline app/views/recipe/create.html.haml]: rm app/views/recipe/*.html.erb

File: spec/controllers/recipe_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+RSpec.describe RecipeController, :type => :controller do
+
+  describe "Get the current user" do
+    login_user
+    it "should have a current_user" do
+    expect(subject.current_user).to_not eq(nil)
```
Comment (moelheni) [inline spec/controllers/recipe_controller_spec.rb]: Indent here

File: spec/factories/users.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+FactoryGirl.define do
+  factory :user do
+    email 'admin@example.com'
```
Comment (moelheni) [inline spec/factories/users.rb]: use Faker::Internet::email

File: spec/models/user_spec.rb (hunk 1 above)
Comment (SamiDghim) [inline spec/models/user_spec.rb]: I still need it to test user model

**Comment by SamiDghim** [comment]
@achr3f @insatmc @melas123 @DoniaMroua @thameursaadi

File: app/helpers/recipe_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module RecipeHelper
```
Comment (achr3f) [inline app/helpers/recipe_helper.rb]: plz delete this file

File: spec/factories/users.rb (hunk 4 above)
Comment (achr3f) [inline spec/factories/users.rb]: :+1:

File: app/helpers/recipe_helper.rb (hunk 5 above)
Comment (SamiDghim) [inline app/helpers/recipe_helper.rb]: Done

**Pull request merged**